docusaurus/docs/guides/saving-forms.md

```diff
@@ -0,0 +1,46 @@
+---
+id: saving-forms
+title: Saving forms
+sidebar_label: Saving forms
+---
+
+`ngx-form-object` provides three hooks that can be utilized to add additional functionalities to a saving process. The hooks will be executed one after another in the following order: `beforeSave`, `save`, `afterSave`. Of this three methods only `save` method is <b>required for saving</b> the form.
+
+### beforeSave()
+Implement this method to execute any action before the actual saving is done. `beforeSave` method gets a `FormStore` instance as an argument and it should return an observable of the same `FormStore`.
+
+One example how you could use this hook is to save model relationships before the original model is saved.
+For example, if you set `userForm.value.department` to a new department that does not yet exist in your database, then you  might want to save it before saving the `User` model. `beforeSave` hook could be used to achieve that.
```

Comment:
```suggestion
For example, if `userForm.value.department` is set to a new department that does not yet exist in the database, then it might be useful to save it before saving the `User` model. `beforeSave` hook could be used to achieve that.
```